lib/src/main/java/io/ably/lib/types/BaseMessage.java

```diff
@@ -78,7 +78,12 @@ public void decode(ChannelOptions opts) throws AblyException {
 					if(!match.matches()) break;
 					String xform = match.group(1).intern();
 					if(xform == "base64") {
-						data = Base64Coder.decode((String)data);
+						try {
+							data = Base64Coder.decode((String) data);
+						}
+						catch (IllegalArgumentException e) {
+							throw AblyException.fromThrowable(e);
```

Comment:
What's going to happen to this exception? I don't think it will propagate to anywhere useful. In this case I think we should do the same as in the case of being unable to decrypt - ie leave the `data` and `encoding` unchanged, and log a message saying that the base64 decoding step failed.